### ISSUE TITLE
Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### ISSUE THREAD

**Issue by Seriosk** [barryvdh/laravel-cors#198]
I have set this up, using v0.9.2 and I am receiving the error in the title. I am using Laravel 5.4 on IIS web server.

Here are my routes:
```
Route::group(['domain' => 'api.domain.uk', 'namespace' => 'Api'], function() {
    Route::group(['middleware' => 'cors', 'prefix' => 'ajax'], function() {
        Route::get('/rooms/{id}/get-locked-status', 'ApiController@getRoomLockStatus');
        Route::any('/rooms/{id}/update-locked-status', 'ApiController@updateRoomLockStatus');
    });
});
```

**Comment by AbdurRehman26** [comment]
Anyone got a solution for this ?

**Comment by Phelms215** [comment]
I have found the same issue and I noticed if I add the **Barryvdh\Cors\ServiceProvider::class** to the app.php I get

`The 'Access-Control-Allow-Origin' header contains multiple values`

Remove it from the middleware group, or let run globally and I get the same message you have. Other way around and you get preflight issues.

**Comment by ryanrca** [comment]
I have the same issue.  I am seeing all the correct headers on the request, but the response does not have the correct headers.  I don't think the request is even making it to the server.  I have not a clue what the solution is.

**Comment by Nagarjuna32** [comment]
I have the same issue. Anyone got a solution for this ?

**Comment by crow1796** [comment]
I have the same issue.

**Comment by alexeycrystal** [comment]
Same here...

**Comment by daniel-farina** [comment]
Same problem here, still looking for a solution.

**Comment by sunsreng** [comment]
services.AddCors(options =>
            {
                options.AddPolicy("MAPI",
                      builder =>
                      {
                          builder.WithOrigins("http://localhost:4200")
                                 .AllowAnyHeader()
                                 .AllowAnyMethod();
                      });
            });

**Comment by chbro** [comment]
same here, flag

**Comment by barryvdh** [comment]
Can you try again with https://github.com/barryvdh/laravel-cors/releases/tag/v0.10.1 ?

Please re-open this issue when the problem still persists.

**Comment by behnamazimi** [comment]
I have this issue with Laravel 5.6 as y server side and axios.

Here are the errors that I face:

> OPTIONS http://example.com/api/v1/city/query 405 (Method Not Allowed)

> Failed to load http://example.com/api/v1/city/query: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://localhost:3000' is therefore not allowed access. The response had HTTP status code 405.

**Comment by sfdsfdsfdesfdsghefasefs** [comment]
same issue here man
please help me...

**Comment by davi5e** [comment]
Same here...

CORS works fine if the AJAX request goes to **example.com** (which happens to be the _APP_URL_ env variable), but it fails as described if the request goes to **api.example.com**.

Also, it's worth noting that this domain has a different route map than the default Laravel installation's. In `app/Providers/RouteServiceProvider.php`:

```php
   protected function mapApiRoutes()
    {
        Route::domain('api.' . parse_url(config('app.url'))['host'])
            ->middleware('api')
            ->namespace($this->namespace)
            ->group(base_path('routes/api.php'));
    }
```

The error message is the same:

> Access to XMLHttpRequest at 'http://api.example.com/' from origin 'http://admin.example.com' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by artmerlop** [comment]
I have this issue too, did anyone solved it?

**Comment by folkevil** [comment]
I'm using Laravel 6 have the same issue, even set allow all headers

**Comment by kenegade** [comment]
I'm having the same issue with Laravel 7.  Locally it works fine from localhost ->dev site.  But going for the live Node.js site to Laravel AP siteI, returns  No 'Access-Control-Allow-Origin' header is present on the requested resource.  Note: it's not that the production URL is not an allowed origin, but rather there is no Access Control header being returned from Laravel.

**Comment by cabassi** [comment]
Same issue here...
It works locally with dev app and dev server but not in production :(

**Comment by barryvdh** [comment]
Are you on version 2.0? Please create a new issue and list all headers.
